src/sage/groups/libgap_mixin.py

```diff
@@ -963,3 +965,128 @@ def is_isomorphic(self, H):
+            True
+
+        Test that elements of resultant list are GAP objects::
+
+            sage: from sage.groups.libgap_mixin import minimum_generating_set
+            sage: G = PermutationGroup([(1,2,3), (2,3), (4,5)])
+            sage: s = minimum_generating_set(G); s
+            [(2,3), (1,3,2)(4,5)]
+            sage: s[0].parent()
+            C library interface to GAP
+    """
+    if not isinstance(group, GapElement):
+        try:
+            group = group._libgap_()
+        except AttributeError:
+            raise NotImplementedError("only implemented for groups that contain a working __libgap__ mehtod")
+
+    if not group.IsFinite().sage():
+        raise NotImplementedError("only implemented for finite groups")
+
+    # A function to check if the group is generated by the given generators or not.
+    def is_groupbygens(group, gens):
+        if not isinstance(group, GapElement):
+            group = group._libgap_()
+        return set(group.AsList()) == set(libgap.GroupByGenerators(gens).AsList())
+
+    group_elements = group.AsList()
+
+    if group.IsCyclic().sage():
+        for ele in group_elements:
+            if is_groupbygens(group, [ele]):
+                return [ele]
+
+    if group.IsSimple().sage():
+        n = len(group_elements)
+        for i in range(n):
+            for j in range(i+1, n):
+                if is_groupbygens(group, [group_elements[i], group_elements[j]]):
+                    return [group_elements[i], group_elements[j]]
+
+    N = group.MinimalNormalSubgroups()[0]
+    n = N.SmallGeneratingSet()
+
+    phi = group.NaturalHomomorphismByNormalSubgroup(N)
+    GbyN = phi.ImagesSource()
+    GbyN_mingenset = minimum_generating_set(GbyN)
+
+    g = [phi.PreImagesRepresentative(g) for g in GbyN_mingenset]
+    l = len(g)
+
+    if N.IsAbelian().sage():
+        if is_groupbygens(group, g):
+            return g
+
+        for i in range(l):
+            for j in range(len(n)):
+                temp = g[i]
+                g[i] = g[i]*n[j]
+                if is_groupbygens(group, g):
+                    return g
+                g[i] = temp
+
+        return g + [n[0]]
+
+    # A function to generate some combinations of the generators
+    # of the group according to the algorithm.
+    def gen_combinations(g, N, t):
+        if t > len(g):
+            t = len(g)
+        if t <= 0:
+            yield g
+        for go in gen_combinations(g,N,t-1):
+            for j in range(len(N)):
+                temp = go[t-1]
+                go[t-1] = go[t-1]*N[j]
+                yield go
+                go[t-1] = temp
+
+    # Here, the value of t is different from the one in the research paper mentioned
+    # in the Algorithm section becuase in the paper the value of t was calculated using
+    # cheif series of the group but currently sage does not have cheif series implemented.
+    t = ceil((13/5 + log(group.Size().sage(), 2) / log(N.Size().sage(), 2)))
```

Comment:
You can access the Cheif series from GAP.

Reply:
Can I do that in another PR? Because calculating t is a big task and will take time. I am also busy making GSOC proposals and working on my coursework.

Reply:
No, it should be done here. Why is it such a big task? It's there in the paper, right? Otherwise you would need to *prove* that your bound is sufficient. It doesn't seem to be in the paper, but I haven't read it super carefully.

Also, it would be good if there was a bit more specific detail about where in the paper you are referencing, both here and in the `ALGORITHM:` section Actually, given the writing in the paper, it probably would be better to sketch out the algorithm in the `ALGORITHM:` section.

Reply:
I have changed the method, and now it does not require computing t; while revisiting the paper for the proof, I noticed that the paper is just proving that there exists some t which limits the number of checks which we have to do. But if we check the combinations in some specific order then we actually do not have to compute t and that is what I have done in this new commit and this is efficient also because it eliminates steps to calculate t.

Reply:
> Also, it would be good if there was a bit more specific detail about where in the paper you are referencing, both here and in the ALGORITHM: section Actually, given the writing in the paper, it probably would be better to sketch out the algorithm in the ALGORITHM: section.

I will complete this by tomorrow.